src/utils.ts

```diff
@@ -2316,10 +2821,111 @@ export class Bar extends newRect {
   }
 }
 
+export class newLabel extends newShape {
+  public shapeSize: Vertex = new Vertex(30, 12);
+  private legend: newRect | LineSegment | newPoint2D;
+  public maxWidth: number = 150;
+  public readonly textOffset = 5;
+  constructor(
+    shape: newShape,
+    public text: newText,
+    public origin: Vertex = new Vertex(0, 0)
+    ) {
+      super();
+      this.isScaled = false;
+      this.text.width = this.maxWidth - this.shapeSize.x;
+      this.getShapeStyle(shape, this.origin);
+      this.buildPath();
+    }
+
+  public buildPath(): void {
+    this.legend.buildPath();
+    this.path = this.legend.path;
+  }
+
+  protected buildUnscaledPath(context: CanvasRenderingContext2D) {
+    const matrix = context.getTransform();
+    context.resetTransform();
+    const origin = this.origin.transform(matrix);
+    this.buildPath();
+    const path = new Path2D();
+    path.addPath(this.path);
+    this.path = new Path2D();
+    this.path.addPath(path, matrix.inverse());
+    this.path.addPath(this.text.path, matrix.inverse());
+    return path
+  }
+
+  private updateLegendGeometry(): void {
+    if (this.legend instanceof LineSegment) {
+      const margin = 2;
+      this.legend.origin.x = this.origin.x;
+      this.legend.origin.y = this.origin.y + margin;
+      this.legend.end.x = this.origin.x + this.shapeSize.x;
+      this.legend.end.y = this.origin.y + this.shapeSize.y - margin;
+    }
+    else if (this.legend instanceof newPoint2D) this.legend.center = this.origin.add(this.shapeSize.divide(2));
+    else this.legend = new newRect(this.origin, this.shapeSize);
+  }
+
+  public getShapeStyle(shape: newShape, origin: Vertex): void {
+    const style = shape.getStyle();
+    if (!shape.isFilled && !(shape instanceof newPoint2D)) this.legend = new LineSegment(origin.copy(), origin.add(this.shapeSize));
+    else if (shape instanceof newPoint2D) {
+      this.legend = new newPoint2D(origin.x, origin.y);
+      this.legend.size = this.shapeSize.y * 0.9;
+      this.legend.marker = shape.marker;
+      this.legend.markerOrientation = shape.markerOrientation;
+    }
+    else this.legend = new newRect(origin.copy(), this.shapeSize);
+    Object.entries(style).map(([key, value]) => this[key] = value);
+  }
+
+  public static deserialize(data: any, scale: Vertex = new Vertex(1, 1)): newLabel {
+    const textParams = newText.deserializeTextParams(data);
+    const shape = data.shape ? newShape.deserialize(data.shape, scale) : new newRect();
+    const text = new newText(data.title, new Vertex(0, 0), textParams);
+    text.isScaled = false;
+    text.baseline = "middle";
+    text.align = "start";
+    const label = new newLabel(shape, text);
+    return label
```

Comment:
You could also directly `return new newLabel(shape, text);` instead of assigning it